base/task.jl

```diff
@@ -56,6 +56,31 @@ function showerror(io::IO, ex::CompositeException)
     end
 end
 
+"""
+    FailedTaskException
+
+This exception is thrown by a `wait(t)` call when task `t` fails.
+`FailedTaskException` wraps the failed task `t`.
+"""
+struct FailedTaskException <: Exception
+    task::Task
+end
+
+function showerror(io::IO, ex::FailedTaskException)
+    stacks = []
+    while isa(ex.task.exception, FailedTaskException)
+        pushfirst!(stacks, ex.task.backtrace)
+        ex = ex.task.exception
+    end
```

Comment:
Ok, this surfaces the underlying error through multiple levels of `wait()`. That's nice:

```julia
julia> t1 = @async sum()
       t2 = @async wait(t1)
       wait(t2)

ERROR: FailedTaskException:
MethodError: no method matching sum()
Closest candidates are:
  sum(::Tuple{Any,Vararg{Any,N} where N}) at tuple.jl:379
  sum(::StepRangeLen) at twiceprecision.jl:536
  sum(::AbstractRange{#s66} where #s66<:Real) at range.jl:978
  ...
Stacktrace:
 [1] (::getfield(Main, Symbol("##23#24")))() at ./task.jl:332
Stacktrace:
 [1] wait(::Task) at ./task.jl:251
 [2] (::getfield(Main, Symbol("##25#26")))() at ./task.jl:332
Stacktrace:
 [1] wait(::Task) at ./task.jl:251
 [2] top-level scope at REPL[5]:3
```